mlir/lib/ExecutionEngine/conv-validation-wrappers.cpp

```diff
@@ -553,14 +554,78 @@ extern "C" void mcpuVerifyFloat(float *gpuAllocated, float *gpuAligned,
 }
 
 // Compare the results in int32
+template <typename VALTYPE>
+void mcpuVerifyInt(int32_t *gpuAligned, VALTYPE *valAligned, int64_t dataSize,
+                   char printDebug) {
+  int64_t failure_count = 0;  // the number of incorrect elements
+  int64_t overflow_count = 0; // the number of overflow elements
+  int64_t maxAbsDiff = 0;
+
+  PrintOption print_option = static_cast<PrintOption>(printDebug);
+  for (int64_t i = 0; i < dataSize; ++i) {
+    int64_t valNum = static_cast<int64_t>(valAligned[i]);
+    int32_t gpuNum = gpuAligned[i];
+    if (valNum > INT32_MAX || valNum < INT32_MIN) {
+      overflow_count++;
+      if (print_option == PrintOption::Always)
```

Comment:
This is `Always || Failure`, I'd think?

Reply:
I'd say overflow indicates bad test vectors for the GPU but not a failure